eox_nelp/notifications/notify_course_due_date.py

```diff
@@ -0,0 +1,78 @@
+"""Notificaty course date module. This manage the delivery of an
+upcoming_course_due_date
+"""
+import logging
+
+from django.utils import timezone
+
+from eox_nelp.edxapp_wrapper.bulk_email import CourseEmailTemplate, get_course_email_context
+from eox_nelp.edxapp_wrapper.student import CourseEnrollment
+from eox_nelp.notifications.utils import (
+    get_xblock_course_from_usage_key_string,
+    get_xblock_from_usage_key_string,
+    send_email_multialternative,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def notify_upcoming_course_due_date(upcoming_course_due_date):
+    """Send the correspongind email of an UpcomingCourseDate object.
+    This is using the bulk-email template that already exists.
+    Args:
+        upcoming_course_due_date (UpcomingCourseDate): UpcomingCourseDate object to sent emails.
+    """
+    #  pylint: disable=too-many-locals
```

Comment:
if pylint raises this warning you should consider to stop using so many variables or split the method

Reply:
[c782c5a](https://github.com/eduNEXT/eox-nelp/pull/44/commits/c782c5a6be0ba58eb4ff465f8c256db32441b5a4)